packages/js/product-editor/src/components/header/hooks/use-publish/use-publish.tsx

```diff
@@ -93,6 +95,12 @@ export function usePublish( {
 							? 'product_publish_error'
 							: 'product_create_error',
 					} as WPError;
+					if ( ( error as Record< string, string > ).variations ) {
+						wpError.code = 'variable_product_no_variation_prices';
+						wpError.message = (
+							error as Record< string, string >
+						 ).variations;
+					}
```

Comment:
This is slightly dirty, but allows us to show a specific message just for the variations part.